source/beamline/CLS/CLSBeamline.cpp

```diff
@@ -1,12 +1,40 @@
 #include "CLSBeamline.h"
 
+#include "beamline/CLS/CLSStorageRing.h"
+
 CLSBeamline::CLSBeamline(const QString &controlName) :
 	AMBeamline(controlName)
 {
-
+	wasBeamAvailable_ = false;
 }
 
 CLSBeamline::~CLSBeamline()
 {
 
 }
+
+/**
+ * ================= implementation of public interface / slots =======================
+ **/
+
+void CLSBeamline::initializeBeamline()
+{
+	AMBeamline::initializeBeamline();
+
+	connect(CLSStorageRing::sr1(), SIGNAL(beamAvaliability(bool)), this, SLOT(updateBeamAvailabilityStatus()));
+
+	updateBeamAvailabilityStatus();
+}
+
+/**
+ * ================= implementation of protected slots / functions =======================
+ **/
+void CLSBeamline::updateBeamAvailabilityStatus()
```

Comment:
Just checking. It looks like this method should be called every time the storage ring beam availability changes and when the beamline beam availability changes. I see a connect statement above for the storage ring, but I'm not sure I see something for the beamline availability?